src/components/plots/rectanglePlot.ts

```diff
@@ -0,0 +1,58 @@
+///<reference path="../../reference.ts" />
+
+module Plottable {
+export module Plot {
+  export class Rectangle<X, Y> extends AbstractXYPlot<X, Y> {
+
+    /**
+     * Constructs a RectanglePlot.
+     *
+     * A RectanglePlot consists of a bunch of rectangles. The user is required to
+     * project the left and right bounds of the rectangle (x1 and x2 respectively)
+     * as well as the bottom and top bounds (y1 and y2 respectively)
+     *
+     * @constructor
+     * @param {Scale.AbstractScale} xScale The x scale to use.
+     * @param {Scale.AbstractScale} yScale The y scale to use.
+     */
+    constructor(xScale: Scale.AbstractScale<X, any>, yScale: Scale.AbstractScale<Y, any>) {
+      super(xScale, yScale);
+      this.classed("rectangle-plot", true);
+    }
+
+    protected _getDrawer(key: string) {
+      return new _Drawer.Rect(key, true);
+    }
+
+    protected _generateAttrToProjector() {
+      var attrToProjector = super._generateAttrToProjector();
+
+      // Copy each of the different projectors.
+      var x1Attr = attrToProjector["x1"];
+      var y1Attr = attrToProjector["y1"];
```

Comment:
x1/x2 are reliant on the projection of x for the calculations to be accurate I believe

https://github.com/palantir/plottable/pull/1711/files#diff-85d9beec29bb331c98aa7092fd9d30d8R62